mimic/test/test_nova.py

```diff
@@ -688,6 +689,92 @@ def test_confirm_and_revert_server_resize(self):
             treq.json_content(reverted_server_response))
         self.assertEqual(reverted_server_response_body['server']['flavor']['id'], '2')
 
+    def test_reboot_server(self):
+        """
+        A hard reboot of a server sets the server status to HARD_REBOOT and returns a 202
+        A soft reboot of a server sets the server status to REBOOT and returns a 202
+        After some amount of time the server will go back to ACTIVE state
+        The clock is being used to advance time and verify that status changes from the
+            a reboot state to active.  The current time interval being used in hardcoded
+            in the route for now. In the future we need to refactor to allow different
+            durations to be set including a zero duration which would allow the server to
+            skip the intermediary state of HARD_REBOOT or REBOOT and go straight to ACTIVE
+        If the 'type' attribute is left out of the request, a response body is returned
+            with code of 400
+        http://docs.rackspace.com/servers/api/v2/cs-devguide/content/Reboot_Server-d1e3371.html
+        """
+        no_reboot_type_request = json.dumps({"reboot": {"missing_type": "SOFT"}})
+        response, body = self.successResultOf(json_request(
+            self, self.root, "POST",
+            self.uri + '/servers/' + self.server_id + '/action', no_reboot_type_request))
+        self.assertEqual(response.code, 400)
+        self.assertEqual(body, {
+            "badRequest": {
+                "message": "Missing argument 'type' for reboot",
+                "code": 400
+            }
+        })
+
+        wrong_reboot_type_request = json.dumps({"reboot": {"type": "FIRM"}})
+        response, body = self.successResultOf(json_request(
+            self, self.root, "POST",
+            self.uri + '/servers/' + self.server_id + '/action', wrong_reboot_type_request))
+        self.assertEqual(response.code, 400)
+        self.assertEqual(body, {
+            "badRequest": {
+                "message": "Argument 'type' for reboot is not HARD or SOFT",
+                "code": 400
+            }
+        })
+
+        # Soft reboot tests
+        soft_reboot_request = json.dumps({"reboot": {"type": "SOFT"}})
+        soft_reboot = request(
+            self, self.root, "POST",
+            self.uri + '/servers/' + self.server_id + '/action', soft_reboot_request)
+        soft_reboot_response = self.successResultOf(soft_reboot)
+        self.assertEqual(soft_reboot_response.code, 202)
+
+        soft_reboot_server = request(
```

Comment:
This is not required here, but in the future you could use `json_request` instead and as it returns a tuple of the response and content. Using that you can skip the 2 `successResultOf` and just have one.